src/saleor_app/security.py

```diff
@@ -0,0 +1,174 @@
+import hashlib
+import hmac
+import logging
+from typing import Optional
+
+import jwt
+from fastapi import HTTPException, Request, Security
+from fastapi.openapi.models import APIKey as APIKeyModel
+from fastapi.openapi.models import APIKeyIn
+from fastapi.security.base import SecurityBase
+from jwt.api_jwk import PyJWKSet
+from pydantic import BaseModel
+from starlette.status import HTTP_401_UNAUTHORIZED
+
+from saleor_app.schemas.core import Saleor
+from saleor_app.schemas.principals import SaleorPrincipal
+
+logger = logging.getLogger(__name__)
+
+SALEOR_DOMAIN_HEADER = "x-saleor-domain"
+SALEOR_TOKEN_HEADER = "x-saleor-token"
+SALEOR_SIGNATURE_HEADER = "x-saleor-signature"
+
+
+class SaleorDomainSecurity(SecurityBase):
+    def __init__(
+        self,
+        *,
+        auto_error: bool = True,
+    ):
+        self.model = APIKeyModel(
+            **{"in": APIKeyIn.header},
```

Comment:
Why u need to unpack it?

Reply:
`in` being a Python keyword, can't be used as an argument, that's how they defined the args there

Reply:
But correct me if I'm wrong -  it does not matter if u use it as an fn keyword argument, right? Ur not shadowing anything